base/ltexpl.dtx

```diff
@@ -516,4 +516,27 @@
 %<latexrelease>\EndIncludeInRelease
 %    \end{macrocode}
 %
+% \DescribeMacro\IfExplAtLeastTF
+%   A pretty simple wrapper.
+%
+% \begin{macro}{\IfExplAtLeastTF}
+% \changes{v1.3g}{2023/10/12}{Provide a test for \pkg{expl3} date (gh/1004)}
+%    \begin{macrocode}
+%<*2ekernel|latexrelease>
+%<latexrelease>\IncludeInRelease{2023/11/01}%
+%<latexrelease>                 {\IfExplAtLeastTF}{Test for expl3 date}%
+\def\IfExplAtLeastTF{\@ifl@t@r\ExplLoaderFileDate}
+%    \end{macrocode}
+% \end{macro}
+%
+%    \begin{macrocode}
+%</2ekernel|latexrelease>
+%<latexrelease>\EndIncludeInRelease
+%<latexrelease>\IncludeInRelease{0000/00/00}%
+%<latexrelease>                 {\IfExplAtLeastTF}{Test for expl3 date}%
+%<latexrelease>
+%<latexrelease>\let\IfExplAtLeastTF\@undefined
+%<latexrelease>\EndIncludeInRelease
```

Comment:
I don't know  what we did concerning rollback in case of \IfFormatAtlLeastTF (would need to check) but is it wise to undefine this command in rollback? I  mean it is intended to be used by packages and those package would then automatically fail in rollback. But we don't roll back the L3 prog layer, so in fact it will in rollback still offer the same  as before.

Thus, I think in this particular case it should either not have a rollback at all or it should have the same definition as line 528 (allowing for a roll forward).

Reply:
FYI, it's pretended that `\IfFormatAtLeastTF`, `\IfPackageAtLeastTF` and similar commands were "available since the beginning of dawn".
https://github.com/latex3/latex2e/blob/6e3a3e37cf74dc101b445ba799aabb9de9cafc82/base/ltclass.dtx#L912-L923

Reply:
The `latexrelease` package was introduced in 2015-01-01, while expl3 was not part of the format untill 2020-02-02. Thus `\ExplLoaderFileDate` may not exist in the format when rolling forward.

Reply:
Well, roll-forward was always rather fragile and only intended to help with situations like Overleaf being 1 or 2 releases behind. Integrating expl3 was a kind of major cut  (as it is not roll back or rolled forward) so I doubt that you can roll forward at all from a format before 2020-02-02 at all, I guess it would be in other places first.